functions.py

```diff
@@ -0,0 +1,192 @@
+''' Shared library of functions for report.py and remind.py
+'''
+
+import os
+import re
+import datetime
+import bugzilla
+from jira import JIRA
+
+
+def generate_html_file(htmlcode, remind=False):
+	''' generates HTML file of reminder
+	'''
+	try:
+		os.makedirs('archive')
+	except FileExistsError:
+		pass
+	if remind:
+		filename = './archive/reminder_{:%m%d%Y_%H:%M:%S}.html'.format(datetime.datetime.now())
+	else:
+		filename = './archive/report_{:%m%d%Y_%H:%M:%S}.html'.format(datetime.datetime.now())
+	with open(filename, 'w') as file:
+		file.write(htmlcode)
+
+
+def get_bugs_dict(bug_ids, config):
+	''' takes in set of bug_ids and returns dictionary with
+		bug_ids as keys and API data as values
+	'''
+
+	# initialize bug dictionary
+	bugs = {}
+
+	# iterate through bug ids from set
+	for bug_id in bug_ids:
+
+		# 0 should be default in YAML file (i.e. no bugs recorded)
+		# if present reference should be made in bug dict
+		if bug_id == 0:
+			bugs[0] = {'bug_name': 'No bug on file', 'bug_url': None}
+			continue
+
+		# get bug info from bugzilla API
+		try:
+
+			# hotfix: API call does not work if '/' present at end of URL string
+			parsed_bz_url = config['bz_url'].rstrip('/')
+
+			bz_api = bugzilla.Bugzilla(parsed_bz_url)
+			bug = bz_api.getbug(bug_id)
+			bug_name = bug.summary
+		except Exception as e:
+			print("Bugzilla API Call Error: ", e)
+			bug_name = "BZ#" + str(bug_id)
+		finally:
+			bug_url = config['bz_url'] + "/show_bug.cgi?id=" + str(bug_id)
+			bugs[bug_id] = {'bug_name': bug_name, 'bug_url': bug_url}
+
+	return bugs
+
+
+def get_bugs_set(blockers):
+	''' takes in blockers object and generates a set of all unique bug ids
+		including 0 if it is present
+	'''
+	bug_list = []
+	for job in blockers:
+		bz = blockers[job]['bz']
+		bug_list.extend(bz)
+
+	return set(bug_list)
+
+
+def get_jenkins_jobs(server, job_search_fields):
+	''' takes in a Jenkins server object and job_search_fields string
+		returns list of jobs with given search field as part of their name
+	'''
+
+	# parse list of search fields
+	fields = job_search_fields.split(',')
+
+	# fetch all jobs from server
+	all_jobs = server.get_jobs()
```

Comment:
I don't see this variable used anywhere else, so you could just iterate over the `server.get_jobs()` list unless you prefer the extra readability this variable serves.

Reply:
Going to keep the var for readability, maybe future use